libsolidity/parsing/Parser.cpp

```diff
@@ -493,6 +522,13 @@ Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(bool _forceEmptyN
 			else
 				result.stateMutability = parseStateMutability();
 		}
+		else if (token == Token::Override)
```

Comment:
I think this should also take `allowModifiers &&`, at least `override` should not be allowed for function types, where `allowModifiers` is set to false. Maybe `allowModifiers` could be renamed.